elephant/phase_analysis.py

```diff
@@ -235,27 +244,35 @@ def pairwise_phase_consistency(phases):
         if phase_array.ndim != 1:
             raise ValueError("Phase arrays must be 1D (use .flatten())")
 
-    result_ppc = []
-
-    for phase_array in phases:
-        n_trials = phase_array.shape[0]
-
-        # Compute the distance between each pair of phases using dot product
-        # Optimize computation time using array multiplications instead of for
-        # loops
-        p_cos_2d = np.tile(np.cos(phase_array), reps=(n_trials, 1))
-        p_sin_2d = np.tile(np.sin(phase_array), reps=(n_trials, 1))
-
-        # By doing the element-wise multiplication of this matrix with its
-        # transpose, we get the distance between phases for all possible pairs
-        # of elements in 'phase'
-        dot_prod = np.multiply(p_cos_2d, p_cos_2d.T) + \
-            np.multiply(p_sin_2d, p_sin_2d.T)
-
-        # Now average over all elements in temp_results (the diagonal are 1
-        # and should not be included)
-        np.fill_diagonal(dot_prod, 0)
-        ppc = 2 / (n_trials * n_trials - n_trials) * np.sum(dot_prod)
-        result_ppc.append(ppc)
-
-    return result_ppc
+    if method not in ['ppc0']:
+        raise ValueError('For method choose out of: ["ppc0"]')
+
+    phase_array = np.hstack(phases)
+    n_trials = phase_array.shape[0]  # 'spikes' are 'trials' as in paper
+
+    # Compute the distance between each pair of phases using dot product
+    # Optimize computation time using array multiplications instead of for
+    # loops
+    p_cos_2d = np.tile(np.cos(phase_array), reps=(n_trials, 1))  # TODO: optimize memory usage
```

Comment:
right, `numpy.broadcast_to` is recommended since it does not consume memory as tile does.

Reply:
I was thinking about specifying datatype, using np.float32 would allready reduce half of the memory used. I noticed when trying this on large spiketrains that it consumes too much RAM (nspikes^2). I don't know if broadcast_to will use less memory than tile, I'm not planning on testing it, unless you think it's worth the time.

Reply:
According to numpy's doc, broadcast_to should use only N memory instead of N^2, as in tile.
Because it does not allocate memory but rather returns a read-only view with the given shape. And it's everywhere stated through the numpy doc that broadcast_to should be preferred for exactly this reason - memory.

Reply:
Therefore, we can use the datatype returned by `cos`, which is double float by default.

Reply:
ah I see, since its a view it wont actually construct a new array. And would we save memory if we add `dtype=np.float32` to `np.multiply` in the following line? Because the `dot_prod` is going to be a new matrix of size N^2 (not a view like cos).

Reply:
ah, yes, so we do need to have N^2 after all.
Then yes, I'd cast everything to float32